### PR TITLE
feat: add authorization audit log (AuditEventSink/AuditingAuthorization)

### PR DESCRIPTION
**Pull request by Ahoo-Wang**
## Summary

Adds a pluggable **audit log** for authorization decisions, following the design in `docs/superpowers/specs/2026-08-17-audit-log-design.md`.

Every `Authorization.authorize()` decision is emitted as a structured `AuditEvent` through the new `AuditEventSink` SPI — one implementation covers webflux/webmvc/gateway and any direct `Authorization` usage, with **zero changes to the integration modules** (decorator pattern, same seam as `TracingAuthorization`).

## What's included

- **cosec-api** `me.ahoo.cosec.api.audit`: `AuditEvent` (20 fields: identity / request / decision / matched policy or role permission / `elapsedNanos`), `AuditDecision` (ALLOW / EXPLICIT_DENY / IMPLICIT_DENY / TOO_MANY_REQUESTS / ERROR), `fun interface AuditEventSink`.
- **cosec-core** `me.ahoo.cosec.audit`:
  - `AuditingAuthorization` — decorator publishing via `doOnNext`/`doOnError`; all signals pass through unchanged; event construction **and** publishing are failure-isolated (audit can never alter an authorization outcome); wall-clock timing from Mono subscription to signal.
  - `AuditEventExtractor` — decision mapping table incl. matched policy/statement or role/permission from `VerifyContext`.
  - `LoggingAuditEventSink` (default) — single-line JSON to logger `me.ahoo.cosec.audit`; DENY/TOO_MANY_REQUESTS→WARN, ERROR→ERROR, ALLOW→DEBUG (default INFO level surfaces only denials); lazy serialization.
- **cosec-spring-boot-starter**: `cosec.audit.enabled` (default `true`, follows `LimiterProperties` pattern), auto-configuration with two classpath-conditional `AuditingAuthorization` bean variants, custom-sink override via `@ConditionalOnMissingBean(AuditEventSink)`, graceful skip when the default `Authorization` bean is replaced.
- **OTel composition**: chain is now `filters → TracingAuthorization → AuditingAuthorization → SimpleAuthorization` when both features are active — exactly one `@Primary` in every scenario (audit × otel × custom-bean matrix verified by tests); behavior unchanged when audit is disabled.
- **Docs**: audit sections added to EN + ZH `getting-started/configuration.md`.

## Test evidence

TDD throughout (every behavior RED-first). Full build: **409 tests, 0 failures** (1 pre-existing skip), detekt clean.

## Notes for reviewers

- Two tiny follow-ups parked as backlog (non-blocking): extractor test cross-null assertion asymmetry; hardening `auditedTracingAuthorization`'s condition to bean-name matching for a pathological hand-registered-bean edge case.

### CODE REVIEW

**Comment by Ahoo-Wang** [comment]
@codex

File: cosec-spring-boot-starter/src/test/kotlin/me/ahoo/cosec/spring/boot/starter/audit/AuditOpenTelemetryCompositionTest.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+package me.ahoo.cosec.spring.boot.starter.audit
+
+import io.mockk.mockk
+import me.ahoo.cosec.api.audit.AuditEventSink
+import me.ahoo.cosec.api.authorization.Authorization
+import me.ahoo.cosec.audit.AuditingAuthorization
+import me.ahoo.cosec.opentelemetry.TracingAuthorization
+import me.ahoo.cosec.spring.boot.starter.authorization.CoSecAuthorizationAutoConfiguration
+import me.ahoo.cosec.spring.boot.starter.opentelemetry.CoSecOpenTelemetryAutoConfiguration
+import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-spring-boot-starter/src/test/kotlin/me/ahoo/cosec/spring/boot/starter/audit/AuditOpenTelemetryCompositionTest.kt]: 已修复：两组 Spring 审计测试的断言均已改为 me.ahoo.test.asserts.assert；Detekt 已通过。

File: cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditEventExtractor.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        return eventOf(request, context, decision, reason, elapsedNanos, verifyContext = context.getVerifyContext())
+    }
+
+    private fun toDecision(context: SecurityContext, result: AuthorizeResult): AuditDecision {
+        if (result.authorized) {
+            return AuditDecision.ALLOW
+        }
+        if (context.getVerifyContext()?.result == VerifyResult.EXPLICIT_DENY) {
+            return AuditDecision.EXPLICIT_DENY
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditEventExtractor.kt]: 已修复：每次授权订阅开始时都会清理旧的 VerifyContext，并新增复用同一 SecurityContext 的回归测试，隐式拒绝不再继承历史匹配信息。

File: cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    override fun authorize(request: Request, context: SecurityContext): Mono<AuthorizeResult> {
+        return Mono.defer {
+            val startNanos = System.nanoTime()
+            Mono.defer { delegate.authorize(request = request, context = context) }
+                .doOnNext { result ->
+                    publishSafely {
+                        AuditEventExtractor.fromDecision(
+                            request = request,
+                            context = context,
+                            result = result,
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt]: 已修复：TokenVerificationException 会保存在 SecurityContext 中，审计层在委托拒绝时记录最终返回给客户端的令牌失败结果。

File: cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    override val delegate: Authorization,
+    private val auditEventSink: AuditEventSink,
+) : Authorization,
+    Delegated<Authorization> {
+
+    override fun authorize(request: Request, context: SecurityContext): Mono<AuthorizeResult> {
+        return Mono.defer {
+            val startNanos = System.nanoTime()
+            Mono.defer { delegate.authorize(request = request, context = context) }
+                .doOnNext { result ->
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt]: 已修复：改用可处理空完成信号的 doOnSuccess；Mono.empty() 会记录 IMPLICIT_DENY，同时保持原始空信号不变。

File: cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditEventExtractor.kt [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        if (AuthorizeResult.IMPLICIT_DENY.reason == result.reason) {
+            return AuditDecision.IMPLICIT_DENY
+        }
+        return AuditDecision.EXPLICIT_DENY
+    }
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditEventExtractor.kt:75]: 已修复：toDecision 在通用拒绝分支前识别 TOO_MANY_REQUESTS，并添加了正常返回限流结果的测试。

File: cosec-spring-boot-starter/src/main/kotlin/me/ahoo/cosec/spring/boot/starter/audit/CoSecAuditAutoConfiguration.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+class CoSecAuditAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean(AuditEventSink::class)
+    fun auditEventSink(): AuditEventSink {
+        return LoggingAuditEventSink()
+    }
+
+    @Bean(AUDITING_AUTHORIZATION_BEAN_NAME)
+    @ConditionalOnBean(name = [CoSecAuthorizationAutoConfiguration.BEAN_NAME])
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-spring-boot-starter/src/main/kotlin/me/ahoo/cosec/spring/boot/starter/audit/CoSecAuditAutoConfiguration.kt]: 已修复：审计装饰器现在要求目标 bean 带有 CoSecAuthorizationAutoConfiguration.AutoConfigured 标记；同名自定义 Authorization 不会触发包装。

File: cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditEventExtractor.kt [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+            null -> Unit
+        }
+        return AuditEvent(
+            timestamp = Instant.now(),
+            tenantId = context.tenant.tenantId,
+            principalId = context.principal.id,
+            authenticated = context.principal.authenticated,
+            roles = context.principal.roles,
+            policies = context.principal.policies,
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditEventExtractor.kt]: 已修复：构建 AuditEvent 时通过 toSet() 快照 roles 与 policies，并添加了源集合后续变更不会影响事件的测试。

File: cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+class AuditingAuthorization(
+    override val delegate: Authorization,
+    private val auditEventSink: AuditEventSink,
+) : Authorization,
+    Delegated<Authorization> {
+
+    override fun authorize(request: Request, context: SecurityContext): Mono<AuthorizeResult> {
+        return Mono.defer {
+            val startNanos = System.nanoTime()
+            context.clearVerifyContext()
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt]: 已在本地修复：每个 Reactor 订阅使用独立的 VerifyContextScope，SimpleAuthorization 将匹配结果写入当前订阅槽，审计提取不再依赖共享 SecurityContext 属性；已添加确定性交错的并发回归测试。待提交并推送后解决此线程。

[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply cosec-core/src/main/kotlin/me/ahoo/cosec/audit/AuditingAuthorization.kt]: 修复已推送：2f84a58。并发回归测试和 cosec-core 全量测试通过，Detekt 通过。
